### PR TITLE
Security Configuration

### PR DESCRIPTION
**Pull request by James-Koval**

### CODE REVIEW

File: src/main/java/ita/softserve/course_evaluation_analytics/repository/impl/UsersRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import ita.softserve.course_evaluation_analytics.mapper.UsersMapper;
+import ita.softserve.course_evaluation_analytics.entity.Users;
+import ita.softserve.course_evaluation_analytics.repository.UsersRepository;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UsersRepositoryImpl implements UsersRepository {
+
+    public JdbcTemplate jdbcTemplate;
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/repository/impl/UsersRepositoryImpl.java]: ```NamedParameterJdcbTemplate```

File: src/main/java/ita/softserve/course_evaluation_analytics/repository/impl/UsersRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    public UsersRepositoryImpl(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public Users findByEmail(String email) {
+        String sql = "SELECT * FROM users u WHERE u.email = ?1";
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/repository/impl/UsersRepositoryImpl.java]: extract query into ```query.properties```

File: src/main/java/ita/softserve/course_evaluation_analytics/configuration/SpringConfig.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+public class SpringConfig {
+
+    @Bean
+    public JdbcTemplate getJdbcTemplate() {
+        return new JdbcTemplate(getDataSource());
+    }
+
+    public DataSource getDataSource() {
+        DriverManagerDataSource dataSource = new DriverManagerDataSource();
+        dataSource.setUrl("jdbc:postgresql://localhost:5432/course_evaluation");
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/configuration/SpringConfig.java]: take those values from env.variable

File: src/main/java/ita/softserve/course_evaluation_analytics/entity/Users.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    private Long id;
+    private String firstName;
+    private String lastName;
+    private String email;
+    private String password;
+    private Set<Role> roles;
+
+//    private Group group;
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/entity/Users.java]: you do not need group, course, etc

**Review by Forfend — CHANGES_REQUESTED** [review]

**Review by yapollo2002 — APPROVED** [review]

**Pull request merged**